program_transformers/src/token/mod.rs

```diff
@@ -19,6 +19,64 @@ use {
     spl_token::state::AccountState,
 };
 
+#[allow(clippy::too_many_arguments)]
+pub async fn upsert_owner_for_token_account<T>(
+    txn_or_conn: &T,
+    id: Vec<u8>,
+    token_account: Vec<u8>,
+    owner: Vec<u8>,
+    delegate: Option<Vec<u8>>,
+    slot: i64,
+    frozen: bool,
+    amount: u64,
+    delegate_amount: i64,
+    token_program: Vec<u8>,
+) -> ProgramTransformerResult<()>
+where
+    T: ConnectionTrait + TransactionTrait,
+{
+    let model = token_accounts::ActiveModel {
+        pubkey: ActiveValue::Set(token_account),
+        mint: ActiveValue::Set(id),
+        delegate: ActiveValue::Set(delegate.clone()),
+        owner: ActiveValue::Set(owner.clone()),
+        frozen: ActiveValue::Set(frozen),
+        delegated_amount: ActiveValue::Set(delegate_amount),
+        token_program: ActiveValue::Set(token_program),
+        slot_updated: ActiveValue::Set(slot),
+        amount: ActiveValue::Set(amount as i64),
+        close_authority: ActiveValue::Set(None),
+    };
+
+    let mut query = token_accounts::Entity::insert(model)
+        .on_conflict(
+            OnConflict::columns([token_accounts::Column::Pubkey])
+                .update_columns([
+                    token_accounts::Column::Mint,
+                    token_accounts::Column::DelegatedAmount,
+                    token_accounts::Column::Delegate,
+                    token_accounts::Column::Amount,
+                    token_accounts::Column::Frozen,
+                    token_accounts::Column::TokenProgram,
+                    token_accounts::Column::Owner,
+                    token_accounts::Column::CloseAuthority,
+                    token_accounts::Column::SlotUpdated,
```

Comment:
NIt: I like to put these in the same order as the model above.  I think it avoids things getting missed in the future since you could actually leave a column out in the OnConflict section and would not get an error.